holonote/annotate/annotator.py

```diff
@@ -97,87 +101,78 @@ class AnnotatorInterface(param.Parameterized):
 
     selected_indices = param.List(default=[], doc="Indices of selected annotations")
 
-    kdim_dtypes = param.Dict(default=None, allow_None=True, doc="""
-       Dictionary of one or two key dimension names to dtypes (e.g. int, float, datetime).""")
-
-    connector = param.ClassSelector(class_=Connector, allow_None=False)
+    spec = param.Dict(default={}, doc="Specification of annotation types")
 
-    annotation_table = param.ClassSelector(class_=AnnotationTable, allow_None=False)
+    fields = param.List(default=["description"], doc="List of fields", constant=True)
 
-    region_types = param.ListSelector(default=['Range'], objects=['Range', 'Point'], doc="""
-       Enabled region types for the current Annotator.""")
+    connector = param.ClassSelector(class_=Connector, allow_None=False)
 
     connector_class = SQLiteDB
 
-    def __init__(self, init=True, **params):
-        if "annotation_table" not in params:
-            params["annotation_table"] = AnnotationTable()
-
-        connector_kws = {'fields':params.pop('fields')} if 'fields' in params else {}
+    def __init__(self, spec, *, init=True, **params):
+        connector_kws = {'fields': params.get('fields')} if 'fields' in params else {}
         connector = params.pop('connector') if 'connector' in params else self.connector_class(**connector_kws)
-        super().__init__(connector=connector, **params)
+
+        spec = self.clean_spec(spec)
+
+        super().__init__(spec=spec, connector=connector, **params)
         self._region = {}
         self._last_region = None
-
+        self.annotation_table = AnnotationTable()
         self.annotation_table.register_annotator(self)
         self.annotation_table.add_schema_to_conn(self.connector)
 
         if init:
             self.load()
 
+    @classmethod
+    def clean_spec(self, input_spec: dict[str, Any]) -> SpecDict:
+        """ Convert spec to a DataFrame with columns: type, region
```

Comment:
Docstring says it returns a dataframe when it is returning a dict.

Reply:
Also, I don't think it is cleaning anything. Maybe it should be called `normalize_spec`?

Reply:
I had it as a DataFrame to start with, but that became too troublesome to work with. 

Fine with the name change.